### PR TITLE
add openshift oauth2 backend support

### PR DESCRIPTION
**Pull request by zetaab**
fixes issue #123

### CODE REVIEW

File: social_core/backends/openshift.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        """Return user details from openshift account"""
+        username = response['metadata']['name']
+        email = response['metadata']['name']
+        return {'username': username, 'email': email}
+
+    def user_data(self, access_token, *args, **kwargs):
+        """Loads user data from service"""
+        # user information
+        headers = {'Authorization': 'Bearer '+access_token}
+        data = requests.get(urljoin(append_slash(self.setting('URL')), 'oapi/v1/users/~'), headers=headers).json()
```
Comment (zetaab) [inline social_core/backends/openshift.py]: this is returning following data:

```
{'metadata': {'selfLink': '/oapi/v1/users/jesse.haka%40company.fi', 'creationTimestamp': '2017-03-23T09:25:48Z', 'resourceVersion': '2242696', 'uid': 'b32f0b88-0faa-11e7-b3e4-fa163e023723', 'name': 'jesse.haka@company.fi'}, 'identities': ['company-AD:jesse.haka@company.fi'], 'groups': None, 'apiVersion': 'v1', 'kind': 'User'}
```

File: social_core/backends/openshift.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return urljoin(append_slash(self.setting('URL')), 'oauth/authorize')
+
+    def get_user_id(self, details, response):
+        return response['metadata']['uid']
+
+    def get_user_details(self, response):
+        """Return user details from openshift account"""
+        username = response['metadata']['name']
+        email = response['metadata']['name']
+        return {'username': username, 'email': email}
```
Comment (zetaab) [inline social_core/backends/openshift.py]: using metadata name in both because openshift api does not return much information about user

**Comment by zetaab** [comment]
@omab could you check this?

**Comment by omab** [comment]
Merged, thanks!

**Pull request merged**